Reorganize and update user setup section (#96)
jhudsl/AnVIL_Book_Instructor_Guide: 04_Running_The_Class.Rmd

In our experience, the process has been reasonably straightforward and quick (approved in a couple of days).

# Set Up Student Accounts

In order for your students to compute on AnVIL, they will need to set up an account and provide you with their username so that you can grant them access to resources (Billing Projects and Workspaces).

If possible, it can be helpful to do this ahead of time, so that you do not need to dedicate class time for it.

<br>

::: {.warning}
Make sure you understand when and where your students can spend money!

The following actions will give your students the ability to spend money on AnVIL:

- Adding them to a **Billing Project** - this will let them create their own Workspaces
- Adding them as **Writers or Owners to a Workspace** - this will let them compute in the Workspace
:::

Our recommendation is to **do most set up ahead of time**:

- Create Student Group (and add Instructors)
- Have students create accounts (and collect usernames)
- Add students to the Student Group
- Give the Student Group access to any Workspaces **as Readers**.  This will let them view the contents but not spend money.

> **Wait until shortly before the students need to compute** to add the Student Group to the Billing Project.  This prevents them from starting something up early and then accruing costs by failing to shut it down.

Reviewer: ```suggestion
**Wait until shortly before the students need to compute** to add the Student Group to the Billing Project. This prevents them from accruing costs on your Billing Project without your guidance.
```

This chapter walks through all of the student setup steps that can safely be done ahead of time.  The final step for activating student billing is covered in the next chapter.

## Create Student Group {#student-group}

Creating a Group for your students will let you easily manage the class as a whole by granting the Group access to Billing Projects and Workspaces.

You will need a unique name for your Group. We suggest a combination of institution-class-role-term (e.g., `jhu-bio101-students-2023FA`). Only letters, numbers, underscores, and dashes are allowed in Group names.

To create a Group:

::::{.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_terra_group_create.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

You now have a unique **Student Group**.

> ### Add Instructors to Group

Reviewer: I think this section is kind of redundant with instructor-group.html#add-instructors-and-tas-to-group - can we combine?

Student-facing instructions for account creation (and several other activities) are provided in the [**Student Instructions**](#student-instructions) section.

## Add Students to Group

Once your students have accounts, you will need to add them to your Student Group as **Members**.

::: {.notice}
In order to add students to your Group, you will need the usernames for the Google accounts your students are using to access AnVIL.
:::

Unfortunately there is currently no way to add members to a Group in bulk, so you will need to add students to the Group one-by-one.

The following instructions explain how to add someone to a Terra Group.  You will want to add students to the Group as **Members**.

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_terra_group_add_member.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

## Add Student Group to Workspace(s)

Now that you have added students to the Group, you can easily share Terra resources with the entire Group.

First, you will need to give your Student Group access to any Workspaces that they will need.

> If you are **using a Public Workspace** (most pre-made content uses Public Workspaces), **you can skip this step** - Public Workspaces are automatically available to everyone on AnVIL.  

Reviewer: Isn't it still nice to share the public workspace so it shows up under "My Workspaces" on students' interface? Super minor point, and not sure if this falls under "too much info"

Author: Oh, can you share a public Workspace?  In that case maybe I'll just delete this paragraph.

Author: I don't seem to be able to share arbitrary public Workspaces.  So if somebody is using one of our Workspaces but wasn't involved in developing it then I don't think it's possible for them to share it, unless they want to clone their own copy...
On that note, cloning isn't hard, if we want to recommend doing that, and it would ensure that it stayed stable for them while they were teaching, so maybe we want to recommend that anyway, in which case we could drop this paragraph.

Opened #99 to follow up

Reviewer: Yeah, you can only share a public workspace if you own (I think).. maybe let's not worry about sharing these :)

If you have **Workspaces of your own** that your students will need to access, you should **add the Student Group to your Workspace as a *Reader***.

- Adding students as a Reader gives them permission to view the Workspace, and (once they have funding) to **clone their own copy**.  This will enable them to work safely in their own space without interfering with each other or overwriting anything in the original Workspace.
- Adding students to your Workspace(s) can safely be done ahead of time; as Readers they can't run computations and run up a bill.

> To add the Student Group to the Workspace, you will need the Group email address, which can be found on the Group management page: https://anvil.terra.bio/#groups

Reviewer: ```suggestion
We recommend adding the Student Group to the Workspace, to avoid adding and managing students individually. You will need the Group email address, which can be found on the Group management page: https://anvil.terra.bio/#groups
```

> The following instructions explain how to add someone to a Terra Workspace.  You will want to add students to the Workspace as **Readers**.

Reviewer: ```suggestion
The following instructions explain how to add someone individually to a Terra Workspace. You will want to add students or the Student Group to the Workspace as **Readers**.
```

:::: {.borrowed_chunk}
```{r, echo = FALSE, results='asis'}
cow::borrow_chapter(
  doc_path = "child/_child_workspace_share.Rmd",
  repo_name = "jhudsl/AnVIL_Template"
)
```
::::

Make sure the newly added Student Group displays "Reader" under "Roles".

> # Teach the Class

Reviewer: Might want to create some spacing here - looks a bit weird with the Large Dropcap. <img width="756" alt="image" src="https://user-images.githubusercontent.com/15618412/229290673-549a969e-3c61-4539-b202-fcd400c0c4ee.png">

Author: @avahoffman what browser were you using, and does it look better now?  I added an extra `<br>`

I'm using chrome and it looked okay to me (not shifted over like your picture), so it's hard for me to know if I fixed it.

Reviewer: Looks good! Apparently I'm the only person who uses Firefox, so feel free to ignore :)

<img width="782" alt="image" src="https://user-images.githubusercontent.com/15618412/229635817-25301ac5-e3c3-4d98-9897-1b6b9857fbbb.png">

This chapter covers things that you will generally do on the day of the class:

<br>

**Before class**

- Add students to Billing Project
- Direct students to clone the Workspace

We generally recommend waiting until shortly before the class to add students to the Billing Project, to prevent them from starting something up early and accruing costs.  *However, if you did not create the Terra Billing Project for your class, you should make sure that either you have been given "Owner" permission for the class Billing Project (so you can add students yourself), or that someone who does is prepared to add students in time for your class to start.*

We typically have students clone the Workspace in the first few minutes of class.  This both helps ensure they won't start up their cloud environments early, and ensures they get the most up-to-date version of the Workspace.

**After class**

- Collect feedback
- Give students any necessary instructions on how to save and export any important files, and how to pause or delete their cloud environments

> ## Before: Add Students to Billing Project

Reviewer: I think we should be opinionated and encourage instructors to use the student group wherever possible. This section could refer to the Student Group and not individual students.
